lib/plugins/aws/deploy/lib/uploadArtifacts.js

```diff
@@ -31,10 +31,11 @@ module.exports = {
       .createHash('sha256')
       .update(JSON.stringify(normCfTemplate))
       .digest('base64');
+    const { deploymentDirectoryPrefix, timestamp } = this.serverless.service.package;
 
     let params = {
       Bucket: this.bucketName,
-      Key: `${this.serverless.service.package.artifactDirectoryName}/${compiledTemplateFileName}`,
+      Key: `${deploymentDirectoryPrefix}/${timestamp}/${compiledTemplateFileName}`,
```

Comment:
We actually cannot change value or remove  `this.serverless.service.package.artifactDirectoryName` property, as it's used by some plugins (just checked that).

To avoid confusion we may stop using internally`this.serverless.service.package.artifactDirectoryName` (it'll be just good to add comment above it's assignment that's left for legacy purposes), and maybe introduce a more accurate:

```javascript
serverless.getProvider('aws').s3CodeArtifactsDirectoryPath = `${prefix}/${service}/${stage}/code-artifacts`
serverless.getProvider('aws').s3DeploymentDirectoryPath = `${prefix}/${service}/${stage}/${timestamp}`
```

Still I would precede such change with a separate PR, that:
- Introduces `serverless.getProvider('aws').s3DeploymentDirectoryPath`
- Replaces `service.package.artifactDirectoryName` usage with `serverless.getProvider('aws').s3DeploymentDirectoryPath`

WDYT?

Reply:
Thanks a lot for checking with plugins. I will apply the change as suggested within a dedicated PR preceding this one.